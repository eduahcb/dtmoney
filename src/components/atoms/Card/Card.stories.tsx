import React, { ReactElement } from 'react'
import styled from 'styled-components'

import { Card } from 'components/atoms/Card'
import { Button } from 'components/atoms/Button'

export default {
  title: 'Components/Card',
  component: Card
}

const AlignCenter = styled.div`
  text-align: center;
`

export const usage = (): ReactElement => (
  <Card>
    <AlignCenter>
      <p>this is a text</p>
      <Button color='primary'>my button</Button>
    </AlignCenter>
  </Card>
)
