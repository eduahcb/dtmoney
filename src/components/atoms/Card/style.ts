import styled from 'styled-components'

export const Card = styled.div`
  background-color: var(--shape-principal);
  padding: 24px 32px;
  border-radius: 4px;
`
