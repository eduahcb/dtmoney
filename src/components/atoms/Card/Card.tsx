import React, { ReactElement, ReactNode } from 'react'

import { Card as StyledCard } from './style'

type CardProps = {
  children: ReactNode
}

export const Card = ({ children }: CardProps): ReactElement => {
  return (
    <StyledCard>{ children }</StyledCard>
  )
}
