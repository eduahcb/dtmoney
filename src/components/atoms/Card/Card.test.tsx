import React from 'react'

import { render, screen } from '@testing-library/react'

import { Card } from 'components/atoms/Card'

describe('Card component', () => {
  test('renders children', () => {
    render(<Card><p>this is a text</p></Card>)

    expect(screen.getByText('this is a text')).toBeInTheDocument()
  })

  test('to match snapshot', () => {
    const { asFragment } = render(<Card><p>this is a text</p></Card>)

    expect(asFragment()).toMatchSnapshot()
  })
})
