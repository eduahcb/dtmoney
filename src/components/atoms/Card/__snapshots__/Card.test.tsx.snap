// Vitest Snapshot v1

exports[`Card component > to match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background-color: var(--shape-principal);
  padding: 24px 32px;
  border-radius: 4px;
}

<div
    class="c0"
  >
    <p>
      this is a text
    </p>
  </div>
</DocumentFragment>
`;
